tests/unit/CourtOrderPoa.spec.ts

```diff
@@ -0,0 +1,160 @@
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import { createLocalVue, mount, Wrapper } from '@vue/test-utils'
+
+import { CourtOrderPoa } from '@/components/CourtOrderPoa'
+import VueRouter from 'vue-router'
+
+Vue.use(Vuetify)
+
+let vuetify = new Vuetify({})
+
+const localVue = createLocalVue()
+localVue.use(VueRouter)
+
+/**
+ * Creates and mounts a component, so that it can be tested.
+ *
+ * @param validate The validation prompt.
+ * @returns a Wrapper<CourtOrderPoa> object with the given parameters.
+ */
+function createComponent (validate: boolean = false): Wrapper<CourtOrderPoa> {
+  return mount(CourtOrderPoa, {
+    propsData: { validate: validate },
+    vuetify,
+    localVue
+  })
+}
+
+describe('Court Order and Plan of Arrangement component', () => {
+  it('Loads the component', async () => {
+    const wrapper: Wrapper<CourtOrderPoa> = createComponent()
+    expect(wrapper.findComponent(CourtOrderPoa).exists()).toBe(true)
+
+    wrapper.destroy()
+  })
+
+  it('validates if poa is selected', async () => {
+    const wrapper: Wrapper<CourtOrderPoa> = createComponent()
+
+    // Verify checkbox is NOT selected
+    expect(wrapper.vm.$data.planOfArrangement).toBe(false)
+
+    // Select checkbox
+    const checkBox = wrapper.find('#plan-of-arrangement-checkbox')
+    checkBox.trigger('click')
+    await Vue.nextTick()
+
+    // Verify checkbox is selected
+    expect(wrapper.vm.$data.planOfArrangement).toBe(true)
+
+    // Prompt validates through prop
+    wrapper.setProps({ validate: true })
+    await Vue.nextTick()
+
+    expect(wrapper.find('#court-num-form').text()).toContain('A Court Order number is required')
+
+    wrapper.destroy()
+  })
+
+  it('does NOT validate if poa is NOT selected', async () => {
+    const wrapper: Wrapper<CourtOrderPoa> = createComponent()
+
+    // Verify checkbox is NOT selected
+    expect(wrapper.vm.$data.planOfArrangement).toBe(false)
+
+    // Prompt validates through prop
+    wrapper.setProps({ validate: true })
+    await Vue.nextTick()
+
+    expect(wrapper.find('#court-num-form').text()).toBe('Court Order Number')
```

Comment:
Can you also check the emitted event(s) here (and in unit tests below) please?

Reply:
For sure. I did in the last test when verifying the `valid` state.